### PR TITLE
feat(helpers): extract messages from API error in case of 400

### PR DESCRIPTION
**Pull request by mkucmus**
Part of work of #507 

covers only helpers package

## Checklist

- [x] I followed code and contributing guidelines
- [x] I wrote all needed tests

### CODE REVIEW

File: packages/helpers/src/error/index.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Get the messages from the API response (array of ShopwareErrors)
+ */
+export function getMessagesFromErrorsArray(
+  errors: ShopwareError[]
+): String[] | undefined {
+  if (!errors || !Array.isArray(errors) || !errors.length) {
+    return;
+  }
```
Comment (patzick) [inline packages/helpers/src/error/index.ts]: would be nicer to always return an array :)
+ return primitive string
```suggestion
export function getMessagesFromErrorsArray(
  errors: ShopwareError[]
): string[] {
  if (!errors?.length || !Array.isArray(errors)) {
    return [];
  }
```

File: packages/helpers/src/error/index.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  // return a message only if detail and pointer propery is provided in upcoming ShopwareError
+  return errors
+    .map(
+      ({ detail, source }) =>
+        detail &&
+        source &&
+        source.pointer &&
+        `${source.pointer.substring(1)}: ${detail}`
+    )
+    .filter(message => message);
```
Comment (patzick) [inline packages/helpers/src/error/index.ts]: this is not doing anything :)
```suggestion
```

Comment (mkucmus) [reply packages/helpers/src/error/index.ts]: the output of .map function is possible of type "undefined" (due to condition) so it's just filtering out all the falsy values :) but now I can see the better solution would be .filter(String) to get only the strings...

File: packages/helpers/src/error/index.ts (hunk 1 above)
Comment (mkucmus) [reply packages/helpers/src/error/index.ts]: sure! it's more elegant :+1:

**Comment by patzick** [comment]
great work! 👍

**Pull request merged**